### ISSUE TITLE
Feature request: Viability of sections, columns, and modules

### ISSUE THREAD

**Issue by wmarsh95** [pojome/elementor#1592]
It would be great to have the feature of turning on or the visibility of the module, columns, and/or section. The Beaver Builder page builder has this ability. Here's info on it: http://kb.wpbeaverbuilder.com/article/83-change-element-visibility-by-user-login-and-capability

**Comment by fastflowapps** [comment]
Yes this is something we were just searching for so we can create some integrations. Can this be added?

**Comment by brodneil** [comment]
Hi, I would also really love to have this feature.  Visibility option per section, widget, or page on elementor for loggedin or loggedout users at least. If possible visibility option for roles (ex. different widget or section visibility for subscriber and author, then the better.  Thanks.

**Comment by nicomollet** [comment]
Interested in this too.
At least make it possible to disable/hide an element in the backend.

If I want to temporarily disable an element, I need to remove it in Elementor current state.

**Comment by maxwellhibbert** [comment]
Need

**Comment by JanSjo** [comment]
need

**Comment by graceac** [comment]
It would be great to have this feature.

**Comment by ThrudUK** [comment]
Would be useful :)

**Comment by Spencemeister** [comment]
Please add another individual to the request list.

**Comment by gangesh** [comment]
+1

**Comment by black-eye** [comment]
+1

**Comment by graham00** [comment]
+1

**Comment by advisario** [comment]
+1

**Comment by mediascite** [comment]
+1

**Comment by maxwellhibbert** [comment]
+ User role conditions too

**Comment by bentalgad** [comment]
+1 must! (Also User role conditions, and other conditions too :)

**Comment by bentalgad** [comment]
Would also be great to have "if field x equal to y" display. and so on...

**Comment by ierpe** [comment]
In need too... Basically something like this : https://fr.wordpress.org/plugins/if-shortcode/

Also the possibility to have 2 completely different layout for the same page depending of the logged in status would be great!

**Comment by RichardUUU** [comment]
+1. ierpe, if they implement this, we should be able to put one "page" in a section viewed by people logged in, and another "page" in a section viewed by people not logged in.

**Comment by StreetDog1971** [comment]
+1

**Comment by RichardUUU** [comment]
I have a workaround for anyone using a membership plugin like Woocommerce Membership:

Create your elementor content, then save it as a template. To display conditionally, use HTML like this:
[wcm_restrict plans="annual-member, monthly-member"]
    [elementor-template id="1234"]
[/wcm_restrict]

Those shortcodes are for Woocommerce Membership. Yours might be different.

**Comment by masterrigel** [comment]
Yes. We need this.

**Comment by maherjoshua** [comment]
+1

**Comment by zachgarner** [comment]
New to elementor.. I'm surprised & disappointed that this isn't already a feature.

**Comment by Vexander** [comment]
Yes please

**Comment by axmasons** [comment]
Yes please!

**Comment by ntercity** [comment]
Thanks RichardUUU

Love the work around. i'm using Hide This by andrezrv for the conditional shortcode with template id.

**Comment by abdielleon** [comment]
+1

**Comment by taropaa** [comment]
@codezz That plugin doesn't always work ant their support is not what I would call responsive. I vote that this should be a core feature properly done by the Elementor team.

**Comment by masterrigel** [comment]
Okay. I Agree. I was thinking of the plugin as a temporary stop gap. That's all.

**Comment by codezz** [comment]
> @codezz That plugin doesn't always work ant their support is not what I would call responsive. I vote that this should be a core feature properly done by the Elementor team.

What issues do you have with the plugin?

**Comment by hzv4424** [comment]
+1

**Comment by evanomondi** [comment]
+1

**Comment by jawhalen8463** [comment]
+1

**Comment by Pehja** [comment]
Why is this closed?

**Comment by taropaa** [comment]
@Pehja not sure why this issue is closed here, but I found a similar one open here: https://github.com/elementor/elementor/issues/4404

**Comment by fracii** [comment]
+1

**Comment by kuwaitman** [comment]
i'm looking for an option to let you view the page and edit it as a logged out user.
so it would be called something like "Display page as [user role]" and the user role would also have logged out guests.
Thank you.

**Comment by perezmunoz** [comment]
Need this feature to display an icon that shows a popup when clicked which contains the login widget. When the user logs in I need to change the icon behaviour to redirect to /my-account. Would deeply appreciate this feature.

**Comment by CordialGit** [comment]
By using this **PHP Snippet**:
`// Add CSS class for logged in and logged out users
add_filter('body_class','moh_logged_in_filter');
function moh_logged_in_filter($classes) {
if( is_user_logged_in() ) {
$classes[] = 'logged-in-condition';
} else {
$classes[] = 'logged-out-condition';
}
// return the $classes array
return $classes;
}`


and this **CSS Snippet**:

`/* Logged in & out conditions */`

`.logged-in-condition .hide-logged-in {
	display: none!important;}`

`.logged-out-condition .hide-logged-out {
	display: none!important;}`

I am able to Hide or Show certain Elementor elements to logged in, or logged out users.

Nonetheless, Visibility Logic should be something that comes in-built into Elementor Core.

Thus, I Vote as +1 to this request.

Regards.

**Comment by nicholaszein** [comment]
Hello!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0, and improved with Elementor Pro 3.20.0! 🥳 

You can now use Display Conditions for widgets based on a variety of conditions, including date, time, user role, logged in/out, and others (more to come too). Check the [documentation for the new Display Conditions](https://elementor.com/help/show-and-hide-elements-on-a-page-display-conditions/) for widgets (Pro).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
